docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_encrypted_with_kms_customer_master_keys_remediation.mdx

```diff
@@ -1,6 +1,207 @@
+       kms_key_id = 'your-kms-key-id'
+       ensure_redshift_encryption(cluster_identifier, kms_key_id)
+   ```
+
+4. **Run the Script**:
+   Execute the script to ensure your Redshift clusters are encrypted with the specified KMS CMK.
+
+   ```bash
+   python ensure_redshift_encryption.py
+   ```
+
+This script will check if the specified Redshift cluster is encrypted and, if not, will enable encryption using the provided KMS CMK. Make sure to replace `'your-cluster-identifier'` and `'your-kms-key-id'` with your actual Redshift cluster identifier and KMS key ID.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "KMS Key ID" column. If this column is empty or the key ID listed is not a KMS CMK, then the Redshift cluster is not encrypted with a KMS CMK.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Redshift clusters in your AWS account. You can do this by using the AWS CLI command `aws redshift describe-clusters`. This command will return a JSON output with all the Redshift clusters.
+
+    ```
+    aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+    ```
+
+2. Once you have the list of Redshift clusters, you can check the encryption settings of each cluster. You can do this by using the AWS CLI command `aws redshift describe-clusters --cluster-identifier <cluster-name>`. Replace `<cluster-name>` with the name of your Redshift cluster. This command will return a JSON output with the details of the specified Redshift cluster.
+
+    ```
+    aws redshift describe-clusters --cluster-identifier <cluster-name> --query 'Clusters[*].[KmsKeyId]' --output text
+    ```
+
+3. If the `KmsKeyId` field is empty or null, it means that the Redshift cluster is not encrypted with a KMS CMK. If the `KmsKeyId` field contains a value, it means that the Redshift cluster is encrypted with a KMS CMK.
+
+4. Repeat step 2 and 3 for each Redshift cluster in your AWS account. This way, you can check the encryption settings of all Redshift clusters in your AWS account.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>